src/annotator/highlighter.js

```diff
@@ -150,19 +144,48 @@ function drawHighlightsAbovePdfCanvas(highlightEl) {
  * @typedef {HTMLElement & HighlightProps} HighlightElement
  */
 
+/**
+ * Return the complete text nodes inside `range`.
+ *
+ * If a range starts or ends part-way through a text node, the nodes are split
+ * and the part inside the range is returned.
+ *
+ * @param {Range} range
+ * @return {Text[]}
+ */
+function wholeTextNodesInRange(range) {
+  const textNodes = [];
+
+  forEachNodeInRange(range, node => {
+    if (node.nodeType !== Node.TEXT_NODE) {
+      return;
+    }
+    let text = /** @type {Text} */ (node);
+
+    if (text === range.startContainer && range.startOffset > 0) {
+      text = text.splitText(range.startOffset);
+    }
+
+    if (text === range.endContainer && range.endOffset < text.data.length) {
+      text.splitText(range.endOffset);
+    }
+
+    textNodes.push(text);
+  });
+
+  return textNodes;
+}
+
 /**
  * Wraps the DOM Nodes within the provided range with a highlight
  * element of the specified class and returns the highlight Elements.
  *
- * @param {NormalizedRange} normedRange - Range to be highlighted.
+ * @param {Range} range - Range to be highlighted.
```

Comment:
Nit: you might consider doing away with the full stop here (I know you inherited it)

Reply:
Done